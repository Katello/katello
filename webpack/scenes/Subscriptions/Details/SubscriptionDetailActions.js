import api, { orgId } from '../../../services/api';
import {
  SUBSCRIPTION_DETAILS_REQUEST,
  SUBSCRIPTION_DETAILS_SUCCESS,
  SUBSCRIPTION_DETAILS_FAILURE,
} from './SubscriptionDetailConstants';
import { apiError } from '../../../move_to_foreman/common/helpers.js';

export const loadSubscriptionDetails = subscriptionId => (dispatch, getState) => {
  dispatch({ type: SUBSCRIPTION_DETAILS_REQUEST });

  return api
    .get(`/organizations/${orgId()}/subscriptions/${subscriptionId}`)
    .then(({ data }) => {
      dispatch({
        type: SUBSCRIPTION_DETAILS_SUCCESS,
        response: data,
      });
    })
    .catch(result => dispatch(apiError(SUBSCRIPTION_DETAILS_FAILURE, result)));
};

export default loadSubscriptionDetails;
