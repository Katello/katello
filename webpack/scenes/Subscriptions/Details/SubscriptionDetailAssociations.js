import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'react-bootstrap';
import { translate as __ } from 'foremanReact/common/I18n';
import helpers from '../../../move_to_foreman/common/helpers.js';

const SubscriptionDetailAssociations = ({ subscriptionDetails }) => {
  const searchQuery = 'subscription_id="%s"'.replace('%s', subscriptionDetails.id);

  return (
    <div>
      <h2>{__('Associations')}</h2>
      <Table striped bordered condensed hover>
        <thead>
          <tr>
            <td><b>{__('Resource')}</b></td>
            <td><b>{__('Quantity')}</b></td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>{__('Content Hosts')}</td>
            <td>
              <a href={helpers.urlWithSearch('content_hosts', searchQuery)}>
                {subscriptionDetails.host_count}
              </a>
            </td>
          </tr>
          <tr>
            <td>{__('Activation Keys')}</td>
            <td>
              <a href={helpers.urlWithSearch('activation_keys', searchQuery)}>
                {subscriptionDetails.activation_keys &&
                  subscriptionDetails.activation_keys.length}
              </a>
            </td>
          </tr>
        </tbody>
      </Table>
    </div>
  );
};

SubscriptionDetailAssociations.propTypes = {
  subscriptionDetails: PropTypes.shape({
    id: PropTypes.number,
    host_count: PropTypes.number,
    activation_keys: PropTypes.array,
  }).isRequired,
};

export default SubscriptionDetailAssociations;
