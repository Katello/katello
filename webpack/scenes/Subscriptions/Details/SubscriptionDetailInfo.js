import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'react-bootstrap';
import subscriptionAttributes from './SubscriptionAttributes';

const SubscriptionDetailInfo = ({ subscriptionDetails }) => {
  const subscriptionLimits = (subDetails) => {
    const limits = [];
    if (subDetails.sockets) {
      limits.push(__('Sockets: %s').replace('%s', subDetails.sockets));
    }
    if (subDetails.cores) {
      limits.push(__('Cores: %s').replace('%s', subDetails.cores));
    }
    if (subDetails.ram) {
      limits.push(__('RAM: %s GB').replace('%s', subDetails.ram));
    }
    if (limits.length > 0) {
      return limits.join(', ');
    }
    return '';
  };

  const subscriptionDetailValue = (subDetails, key) => (subDetails[key] == null ? '' : String(subDetails[key]));

  const formatInstanceBased = (subDetails) => {
    if (subDetails.instance_multiplier == null ||
        subDetails.instance_multiplier === '' ||
        subDetails.instance_multiplier === 0) {
      return __('No');
    }
    return __('Yes');
  };

  return (
    <div>
      <h2>{__('Subscription Info')}</h2>
      <Table>
        <tbody>
          {Object.keys(subscriptionAttributes).map(key => (
            <tr key={key}>
              <td><b>{__(subscriptionAttributes[key])}</b></td>
              <td>{subscriptionDetailValue(subscriptionDetails, key)}</td>
            </tr>
            ))}
          <tr>
            <td><b>{__('Limits')}</b></td>
            <td>{subscriptionLimits(subscriptionDetails)}</td>
          </tr>
          <tr>
            <td><b>{__('Instance-based')}</b></td>
            <td>{formatInstanceBased(subscriptionDetails)}</td>
          </tr>
        </tbody>
      </Table>

    </div>
  );
};

SubscriptionDetailInfo.propTypes = {
  subscriptionDetails: PropTypes.shape({}).isRequired,
};

export default SubscriptionDetailInfo;
