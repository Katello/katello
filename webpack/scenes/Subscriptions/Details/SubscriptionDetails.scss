.scrolld-list {
  max-height: 400px;
  overflow-y: auto; 
}

.list-view-pf-expand {
  max-height: 20px;
  padding: 0;
}
