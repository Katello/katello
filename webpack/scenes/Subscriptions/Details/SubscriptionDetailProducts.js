import React from 'react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';
import { ListGroup, ListGroupItem } from 'patternfly-react';
import './SubscriptionDetails.scss';

const SubscriptionDetailProducts = ({ subscriptionDetails }) => (
  <div>
    <h2>{__('Provided Products')}</h2>
    <ListGroup className="scrolld-list">
      {subscriptionDetails.provided_products &&
        subscriptionDetails.provided_products.map(prod => (
          <ListGroupItem key={prod.id}> {prod.name} </ListGroupItem>
          ))}
    </ListGroup>
  </div>
);

SubscriptionDetailProducts.propTypes = {
  subscriptionDetails: PropTypes.shape({
    // Disabling rule as existing code failed due to an eslint-plugin-react update
    // eslint-disable-next-line react/forbid-prop-types
    provided_products: PropTypes.array,
  }).isRequired,
};

export default SubscriptionDetailProducts;
