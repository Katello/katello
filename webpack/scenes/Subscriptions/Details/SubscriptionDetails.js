import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Nav, NavItem, TabPane, TabContent, TabContainer, Grid, Row, Col } from 'patternfly-react';
import BreadcrumbsBar from 'foremanReact/components/BreadcrumbBar';
import SubscriptionDetailInfo from './SubscriptionDetailInfo';
import SubscriptionDetailAssociations from './SubscriptionDetailAssociations';
import SubscriptionDetailProducts from './SubscriptionDetailProducts';
import SubscriptionDetailEnabledProducts from './SubscriptionDetailEnabledProducts';
import { LoadingState } from '../../../move_to_pf/LoadingState';
import { notify } from '../../../move_to_foreman/foreman_toast_notifications';
import api from '../../../services/api';

class SubscriptionDetails extends Component {
  constructor() {
    super();
    this.handleBreadcrumbSwitcherItem = this.handleBreadcrumbSwitcherItem.bind(this);
  }
  componentDidMount() {
    // eslint-disable-next-line react/prop-types
    const routerParams = this.props.match.params;
    this.props.loadSubscriptionDetails(parseInt(routerParams.id, 10));
    this.props.loadProducts({
      subscription_id: parseInt(routerParams.id, 10),
      include_available_content: true,
      enabled: true,
    });
  }

  componentDidUpdate(prevProps) {
    const routerParams = this.props.match.params;
    if (routerParams.id !== prevProps.match.params.id) {
      this.props.loadSubscriptionDetails(parseInt(routerParams.id, 10));
      this.props.loadProducts({
        subscription_id: parseInt(routerParams.id, 10),
        include_available_content: true,
        enabled: true,
      });
    }
  }

  handleBreadcrumbSwitcherItem(e, url) {
    this.props.history.push(url);
    e.preventDefault();
  }

  render() {
    const { subscriptionDetails } = this.props;


    const resource = {
      nameField: 'name',
      resourceUrl: api.getApiUrl('/subscriptions'),
      switcherItemUrl: '/subscriptions/:id',
    };

    if (subscriptionDetails.error) {
      notify({ message: subscriptionDetails.error });
    }

    return (
      <div>
        {!subscriptionDetails.loading && <BreadcrumbsBar
          onSwitcherItemClick={(e, url) => this.handleBreadcrumbSwitcherItem(e, url)}
          data={{
            isSwitchable: true,
            breadcrumbItems: [
              {
                caption: __('Subscriptions'),
                onClick: () =>
                  this.props.history.push('/subscriptions'),
              },
              {
                caption: String(subscriptionDetails.name),
              },
            ],
            resource,
          }}
        />}

        <TabContainer id="subscription-tabs-container" defaultActiveKey={1}>
          <div>
            <LoadingState loading={subscriptionDetails.loading} loadingText={__('Loading')}>
              <Nav bsClass="nav nav-tabs">
                <NavItem eventKey={1}>
                  <div>{__('Details')}</div>
                </NavItem>
                <NavItem eventKey={2}>
                  <div>{__('Enabled Products')}</div>
                </NavItem>
              </Nav>
              <Grid bsClass="container-fluid">
                <TabContent animation={false}>
                  <TabPane eventKey={1}>
                    <div>
                      <Row>
                        <Col sm={6}>
                          <SubscriptionDetailInfo
                            subscriptionDetails={subscriptionDetails}
                          />
                        </Col>
                        <Col sm={6}>
                          <SubscriptionDetailAssociations
                            subscriptionDetails={subscriptionDetails}
                          />
                          <SubscriptionDetailProducts
                            subscriptionDetails={subscriptionDetails}
                          />
                        </Col>
                      </Row>
                    </div>
                  </TabPane>

                  <TabPane eventKey={2}>
                    <div>
                      <Row>
                        <Col sm={12}>
                          <SubscriptionDetailEnabledProducts
                            enabledProducts={subscriptionDetails.enabledProducts}
                          />
                        </Col>
                      </Row>
                    </div>
                  </TabPane>
                </TabContent>
              </Grid>
            </LoadingState>
          </div>
        </TabContainer>
      </div>
    );
  }
}

SubscriptionDetails.propTypes = {
  loadSubscriptionDetails: PropTypes.func.isRequired,
  loadProducts: PropTypes.func.isRequired,
  subscriptionDetails: PropTypes.shape({}).isRequired,
  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
};

export default SubscriptionDetails;
