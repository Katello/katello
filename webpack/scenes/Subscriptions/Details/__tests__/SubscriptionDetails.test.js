import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SubscriptionDetails from '../SubscriptionDetails';
import SubscriptionDetailAssociations from '../SubscriptionDetailAssociations';
import SubscriptionDetailInfo from '../SubscriptionDetailInfo';
import SubscriptionDetailProducts from '../SubscriptionDetailProducts';
import { loadSubscriptionDetails } from '../SubscriptionDetailActions';
import { successState } from './subscriptionDetails.fixtures';

jest.mock('../../../../move_to_foreman/foreman_toast_notifications');

describe('subscriptions details page', () => {
  it('should render and contain appropiate components', async () => {
    const match = { params: { id: 1 } };
    const noop = () => {};

    const wrapper = shallow(<SubscriptionDetails
      loadSubscriptionDetails={loadSubscriptionDetails}
      subscriptionDetails={successState}
      history={{ push: noop }}
      match={match}
    />);
    expect(wrapper.find(SubscriptionDetailAssociations)).toHaveLength(1);
    expect(wrapper.find(SubscriptionDetailInfo)).toHaveLength(1);
    expect(wrapper.find(SubscriptionDetailProducts)).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
