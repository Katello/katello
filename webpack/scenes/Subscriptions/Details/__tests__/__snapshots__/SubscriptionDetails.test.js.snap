// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscriptions details page should render and contain appropiate components 1`] = `
<Grid
  bsClass="container-fluid"
  componentClass="div"
  fluid={false}
>
  <div>
    <Spinner
      className=""
      inline={false}
      inverse={false}
      loading={false}
      size="md"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          sm={12}
        >
          <h1>
            OpenShift Employee Subscription
          </h1>
        </Col>
      </Row>
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          sm={6}
        >
          <SubscriptionDetailInfo
            subscriptionDetails={
              Object {
                "account_number": 1212729,
                "activation_keys": Array [],
                "arch": "ia64,ppc,ppc64,ppc64le,s390,s390x,x86,x86_64",
                "available": 1,
                "consumed": 0,
                "contract_number": 10126160,
                "cores": 4,
                "cp_id": "4028f92a6317cfbd0163b419377f3bee",
                "description": "OpenShift Enterprise",
                "end_date": "2021-12-31 23:59:59 -0500",
                "error": null,
                "host_count": 0,
                "id": 48,
                "instance_multiplier": 1,
                "loading": false,
                "multi_entitlement": true,
                "name": "OpenShift Employee Subscription",
                "product_id": "SER0421",
                "product_name": "OpenShift Employee Subscription",
                "provided_products": Array [
                  Object {
                    "id": 1,
                    "name": "Red Hat OpenShift Container Platform",
                  },
                  Object {
                    "id": 2,
                    "name": "Oracle Java for RHEL Server",
                  },
                  Object {
                    "id": 3,
                    "name": "Red Hat OpenShift Enterprise JBoss EAP add-on",
                  },
                  Object {
                    "id": 4,
                    "name": "Red Hat CloudForms Beta",
                  },
                  Object {
                    "id": 5,
                    "name": "Red Hat CloudForms",
                  },
                  Object {
                    "id": 6,
                    "name": "Red Hat OpenShift Enterprise Client Tools",
                  },
                  Object {
                    "id": 7,
                    "name": "Red Hat Enterprise Linux Atomic Host",
                  },
                  Object {
                    "id": 8,
                    "name": "JBoss Enterprise Application Platform",
                  },
                  Object {
                    "id": 9,
                    "name": "Red Hat JBoss AMQ Clients",
                  },
                  Object {
                    "id": 10,
                    "name": "Red Hat Beta",
                  },
                  Object {
                    "id": 11,
                    "name": "Red Hat OpenShift Enterprise Infrastructure",
                  },
                  Object {
                    "id": 12,
                    "name": "Red Hat Enterprise Linux Fast Datapath Beta",
                  },
                  Object {
                    "id": 13,
                    "name": "Red Hat Ansible Engine",
                  },
                  Object {
                    "id": 14,
                    "name": "Red Hat OpenShift Enterprise Application Node",
                  },
                  Object {
                    "id": 15,
                    "name": "Red Hat OpenShift Enterprise JBoss FUSE add-on",
                  },
                  Object {
                    "id": 16,
                    "name": "Red Hat Software Collections Beta for RHEL Server",
                  },
                  Object {
                    "id": 17,
                    "name": "Red Hat Software Collections for RHEL Server",
                  },
                  Object {
                    "id": 18,
                    "name": "Red Hat Enterprise Linux Fast Datapath",
                  },
                  Object {
                    "id": 19,
                    "name": "Red Hat Enterprise Linux Server",
                  },
                  Object {
                    "id": 20,
                    "name": "Red Hat OpenShift Enterprise JBoss A-MQ add-on",
                  },
                  Object {
                    "id": 21,
                    "name": "JBoss Enterprise Web Server",
                  },
                  Object {
                    "id": 22,
                    "name": "Red Hat JBoss Core Services",
                  },
                ],
                "quantity": 1,
                "ram": null,
                "sockets": null,
                "stacking_id": "SER0421",
                "start_date": "2013-03-01 00:00:00 -0500",
                "subscription_id": 3,
                "support_level": "Self-Support",
                "support_type": "L1-L3",
                "type": "NORMAL",
                "unmapped_guest": false,
                "upstream": true,
                "virt_only": false,
                "virt_who": false,
              }
            }
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          sm={6}
        >
          <SubscriptionDetailAssociations
            subscriptionDetails={
              Object {
                "account_number": 1212729,
                "activation_keys": Array [],
                "arch": "ia64,ppc,ppc64,ppc64le,s390,s390x,x86,x86_64",
                "available": 1,
                "consumed": 0,
                "contract_number": 10126160,
                "cores": 4,
                "cp_id": "4028f92a6317cfbd0163b419377f3bee",
                "description": "OpenShift Enterprise",
                "end_date": "2021-12-31 23:59:59 -0500",
                "error": null,
                "host_count": 0,
                "id": 48,
                "instance_multiplier": 1,
                "loading": false,
                "multi_entitlement": true,
                "name": "OpenShift Employee Subscription",
                "product_id": "SER0421",
                "product_name": "OpenShift Employee Subscription",
                "provided_products": Array [
                  Object {
                    "id": 1,
                    "name": "Red Hat OpenShift Container Platform",
                  },
                  Object {
                    "id": 2,
                    "name": "Oracle Java for RHEL Server",
                  },
                  Object {
                    "id": 3,
                    "name": "Red Hat OpenShift Enterprise JBoss EAP add-on",
                  },
                  Object {
                    "id": 4,
                    "name": "Red Hat CloudForms Beta",
                  },
                  Object {
                    "id": 5,
                    "name": "Red Hat CloudForms",
                  },
                  Object {
                    "id": 6,
                    "name": "Red Hat OpenShift Enterprise Client Tools",
                  },
                  Object {
                    "id": 7,
                    "name": "Red Hat Enterprise Linux Atomic Host",
                  },
                  Object {
                    "id": 8,
                    "name": "JBoss Enterprise Application Platform",
                  },
                  Object {
                    "id": 9,
                    "name": "Red Hat JBoss AMQ Clients",
                  },
                  Object {
                    "id": 10,
                    "name": "Red Hat Beta",
                  },
                  Object {
                    "id": 11,
                    "name": "Red Hat OpenShift Enterprise Infrastructure",
                  },
                  Object {
                    "id": 12,
                    "name": "Red Hat Enterprise Linux Fast Datapath Beta",
                  },
                  Object {
                    "id": 13,
                    "name": "Red Hat Ansible Engine",
                  },
                  Object {
                    "id": 14,
                    "name": "Red Hat OpenShift Enterprise Application Node",
                  },
                  Object {
                    "id": 15,
                    "name": "Red Hat OpenShift Enterprise JBoss FUSE add-on",
                  },
                  Object {
                    "id": 16,
                    "name": "Red Hat Software Collections Beta for RHEL Server",
                  },
                  Object {
                    "id": 17,
                    "name": "Red Hat Software Collections for RHEL Server",
                  },
                  Object {
                    "id": 18,
                    "name": "Red Hat Enterprise Linux Fast Datapath",
                  },
                  Object {
                    "id": 19,
                    "name": "Red Hat Enterprise Linux Server",
                  },
                  Object {
                    "id": 20,
                    "name": "Red Hat OpenShift Enterprise JBoss A-MQ add-on",
                  },
                  Object {
                    "id": 21,
                    "name": "JBoss Enterprise Web Server",
                  },
                  Object {
                    "id": 22,
                    "name": "Red Hat JBoss Core Services",
                  },
                ],
                "quantity": 1,
                "ram": null,
                "sockets": null,
                "stacking_id": "SER0421",
                "start_date": "2013-03-01 00:00:00 -0500",
                "subscription_id": 3,
                "support_level": "Self-Support",
                "support_type": "L1-L3",
                "type": "NORMAL",
                "unmapped_guest": false,
                "upstream": true,
                "virt_only": false,
                "virt_who": false,
              }
            }
          />
          <SubscriptionDetailProducts
            subscriptionDetails={
              Object {
                "account_number": 1212729,
                "activation_keys": Array [],
                "arch": "ia64,ppc,ppc64,ppc64le,s390,s390x,x86,x86_64",
                "available": 1,
                "consumed": 0,
                "contract_number": 10126160,
                "cores": 4,
                "cp_id": "4028f92a6317cfbd0163b419377f3bee",
                "description": "OpenShift Enterprise",
                "end_date": "2021-12-31 23:59:59 -0500",
                "error": null,
                "host_count": 0,
                "id": 48,
                "instance_multiplier": 1,
                "loading": false,
                "multi_entitlement": true,
                "name": "OpenShift Employee Subscription",
                "product_id": "SER0421",
                "product_name": "OpenShift Employee Subscription",
                "provided_products": Array [
                  Object {
                    "id": 1,
                    "name": "Red Hat OpenShift Container Platform",
                  },
                  Object {
                    "id": 2,
                    "name": "Oracle Java for RHEL Server",
                  },
                  Object {
                    "id": 3,
                    "name": "Red Hat OpenShift Enterprise JBoss EAP add-on",
                  },
                  Object {
                    "id": 4,
                    "name": "Red Hat CloudForms Beta",
                  },
                  Object {
                    "id": 5,
                    "name": "Red Hat CloudForms",
                  },
                  Object {
                    "id": 6,
                    "name": "Red Hat OpenShift Enterprise Client Tools",
                  },
                  Object {
                    "id": 7,
                    "name": "Red Hat Enterprise Linux Atomic Host",
                  },
                  Object {
                    "id": 8,
                    "name": "JBoss Enterprise Application Platform",
                  },
                  Object {
                    "id": 9,
                    "name": "Red Hat JBoss AMQ Clients",
                  },
                  Object {
                    "id": 10,
                    "name": "Red Hat Beta",
                  },
                  Object {
                    "id": 11,
                    "name": "Red Hat OpenShift Enterprise Infrastructure",
                  },
                  Object {
                    "id": 12,
                    "name": "Red Hat Enterprise Linux Fast Datapath Beta",
                  },
                  Object {
                    "id": 13,
                    "name": "Red Hat Ansible Engine",
                  },
                  Object {
                    "id": 14,
                    "name": "Red Hat OpenShift Enterprise Application Node",
                  },
                  Object {
                    "id": 15,
                    "name": "Red Hat OpenShift Enterprise JBoss FUSE add-on",
                  },
                  Object {
                    "id": 16,
                    "name": "Red Hat Software Collections Beta for RHEL Server",
                  },
                  Object {
                    "id": 17,
                    "name": "Red Hat Software Collections for RHEL Server",
                  },
                  Object {
                    "id": 18,
                    "name": "Red Hat Enterprise Linux Fast Datapath",
                  },
                  Object {
                    "id": 19,
                    "name": "Red Hat Enterprise Linux Server",
                  },
                  Object {
                    "id": 20,
                    "name": "Red Hat OpenShift Enterprise JBoss A-MQ add-on",
                  },
                  Object {
                    "id": 21,
                    "name": "JBoss Enterprise Web Server",
                  },
                  Object {
                    "id": 22,
                    "name": "Red Hat JBoss Core Services",
                  },
                ],
                "quantity": 1,
                "ram": null,
                "sockets": null,
                "stacking_id": "SER0421",
                "start_date": "2013-03-01 00:00:00 -0500",
                "subscription_id": 3,
                "support_level": "Self-Support",
                "support_type": "L1-L3",
                "type": "NORMAL",
                "unmapped_guest": false,
                "upstream": true,
                "virt_only": false,
                "virt_who": false,
              }
            }
          />
        </Col>
      </Row>
    </Spinner>
  </div>
</Grid>
`;
